src/api/PDFDocument.ts

```diff
@@ -701,27 +708,94 @@ export default class PDFDocument {
     return pdfImage;
   }
 
+  /**
+   * Embed a PDF page into this document.
+   *
+   * For example:
+   * ```js
+   * const pdfDoc = await PDFDocument.create();
+   *
+   * const sourcePdfUrl = 'https://pdf-lib.js.org/assets/with_large_page_count.pdf';
+   * const sourceBuffer = await fetch(sourcePdfUrl).then((res) => res.arrayBuffer());
+   * const sourcePdfDoc = await PDFDocument.load(sourceBuffer);
+   *
+   * // embed page 74 into pdfDoc
+   * const embeddedPage = await pdfDoc.embedPdfDocument(sourcePdfDoc,73);
+   * ```
+   *
+   * @param document The source PDF document that contains the page to be embedded
+   * @param pageIndex Optionally, the page index (starting at 0 for the first page) of the page to embed. Defaults to 0 (the first page).
+   * @returns Resolves with the embedded pdf page.
+   */
+  async embedPdfDocument(
+    document: PDFDocument,
+    pageIndex?: number,
+  ): Promise<EmbeddedPDFPage> {
+    assertIs(document, 'document', [[PDFDocument, 'PDFDocument']]);
+    const page = document.getPages()[pageIndex || 0];
+    const embeddedPage = await this.embedPdfPage(page);
+    return embeddedPage;
+  }
+
+  /**
+   * Embed a PDF page into this document.
+   *
+   * For example:
+   * ```js
+   * const pdfDoc = await PDFDocument.create();
+   *
+   * const sourcePdfUrl = 'https://pdf-lib.js.org/assets/with_large_page_count.pdf';
+   * const sourceBuffer = await fetch(sourcePdfUrl).then((res) => res.arrayBuffer());
+   * const sourcePdfDoc = await PDFDocument.load(sourceBuffer);
+   * const sourcePdfPage = sourcePdfDoc.getPages()[73];
+   *
+   * const embeddedPage = await pdfDoc.embedPdfPage(
+   *   sourcePdfPage,
+   *   { // clip the PDF page to a certain area within the page
+   *     left: 100,
+   *     right: 450,
+   *     bottom: 330,
+   *     top: 570
+   *   },
+   *   [1, 0, 0, 1, 10, 200] // translate by (10,200) units
+   * );
+   * ```
+   *
+   * @param page The source PDF page to be embedded
+   * @param boundingBox Optionally, area of the source page that should be embedded
+   * @param transformationMatrix Optionally, a transformation matrix that is always applied to the embedded page
+   * @returns Resolves with the embedded pdf page.
+   */
+  async embedPdfPage(
+    page: PDFPage,
+    boundingBox?: BoundingBox,
+    transformationMatrix?: TransformationMatrix,
+  ): Promise<EmbeddedPDFPage> {
```

Comment:
I'd like to rename this method to `embedPage`, just for the sake of brevity. Also, please see my comment above about accepting and returning arrays, as it applies to this method as well.